### PR TITLE
k8s: improve jsonpath api to allow both extraction and injection

### PR DESCRIPTION
**Pull request by nicks**
Hello @landism, @maiamcc,

Please review the following commits I made in branch nicks/jsonpath:

3426e7f199012fdab1267f1f9c7c7579a55ae86d (2020-06-18 15:31:42 -0400)
k8s: improve jsonpath api to allow both extraction and injection

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
sorry, this turned out to be a bigger pain in the ass than I expected...in go, map field values are not addressable, so i had to introduce a new struct to do this

File: internal/k8s/jsonpath/value.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package jsonpath
+
+import "reflect"
+
+// A hack over reflect.Value that allows maps keys to be settable.
```
Comment (landism) [inline internal/k8s/jsonpath/value.go:5]: I first read this as "lets you change the keys in a map" (like, change `{"foo": true}` to `{"bar": true}`), which I don't think is what it's doing - would "allows map values to be settable" or "allows values indexed by maps' keys to be settable" work?

Comment (nicks) [reply internal/k8s/jsonpath/value.go]: good point, clarified the language here a bit

**Comment by nicks** [comment]
good point, i missed the tests! copied them over on a separate PR

**Pull request merged**